egs/grid/VSR/conformer_ctc2/train.py

```
@@ -0,0 +1,1078 @@
+
+    return saved_params
+
+
+def save_checkpoint(
+    params: AttributeDict,
+    model: Union[nn.Module, DDP],
+    model_avg: Optional[nn.Module] = None,
+    optimizer: Optional[torch.optim.Optimizer] = None,
+    scheduler: Optional[LRSchedulerType] = None,
+    sampler: Optional[CutSampler] = None,
+    scaler: Optional["GradScaler"] = None,
```

Ruff (0.15.11)

[error] Undefined name `GradScaler`

(F821)